### PR TITLE
Make sure we catch exceptions from mono_class_from_name

### PR DESCRIPTION
**Pull request by peterhall-unity3d**
Change to avoid throwing from mono_class_from_name and instead return NULL.

### CODE REVIEW

**Comment by joncham** [comment]
Looks good @peterhall-unity3d !

Even though this doesn't touch a managed object directly, this seems like another good candidate to move to managed @mrvoorhe @UnityAlex

File: src/coreclr/vm/mono/mono_coreclr.cpp [hunk 1]
```diff
@@ -486,12 +486,25 @@ MonoClass * mono_class_from_name(MonoImage *image, const char* name_space, const
     while (fullTypeName.Find(i, W('/')))
         fullTypeName.Replace(i, W('+'));
 
-    TypeHandle retTypeHandle = TypeName::GetTypeManaged(fullTypeName.GetUnicode(), domainAssembly, FALSE, ignoreCase, TRUE, NULL, NULL);
+    TypeHandle retTypeHandle;
+
+    EX_TRY
+    {
+        retTypeHandle = TypeName::GetTypeManaged(fullTypeName.GetUnicode(), domainAssembly, FALSE, ignoreCase, TRUE, NULL, NULL);
```
Comment (joncham) [inline src/coreclr/vm/mono/mono_coreclr.cpp:493]: Intersting that this still throws an exception when we pass `FALSE` as the 3rd parameter: https://github.com/Unity-Technologies/runtime/blob/unity-main/src/coreclr/vm/typeparse.h#L313

**Review by joncham — APPROVED** [review]

**Comment by joncham** [comment]
> Looks good @peterhall-unity3d !
> 
> Even though this doesn't touch a managed object directly, this seems like another good candidate to move to managed @mrvoorhe @UnityAlex

I am fine landing as is, but doing this in managed if people are available might be a good exercise to do with @peterhall-unity3d

**Comment by UnityAlex** [comment]
> Looks good @peterhall-unity3d !
> 
> Even though this doesn't touch a managed object directly, this seems like another good candidate to move to managed @mrvoorhe @UnityAlex

It does... which impacts what I'm currently working on. I guess I can handle it in my current work. Scope creep approved!

**Pull request merged**